packages/wrangler/src/config/index.ts

```diff
@@ -48,9 +48,11 @@ export function readConfig<CommandArgs>(
 		configPath,
 		args
 	);
-
 	if (diagnostics.hasWarnings()) {
-		logger.warn(diagnostics.renderWarnings());
+		//if json flag is passed and false, or no json flag passed, print warnings
+		if (("json" in args && !args["json"]) || !("json" in args)) {
```

Comment:
Do we really want to do this check here?
I feel like if you don't want warnings then you should set the WRANGLER_LOG to `error` to avoid the `logger.warn()` actually writing to console?
Alternatively, perhaps we could change `logger.warn()` to write to `console.error()` so that it goes to stderr rather than stdout?

Reply:
Another alternative... I note that the command in question (d1 execute) is setup as `withConfig(...)`. But what that does is a call to `readConfig()` which is where these warnings are shown.
Perhaps we could remove that call and manually read the config, having configured the logger appropriately to avoid unwanted warnings?

Reply:
Interesting, i'll give that a go